app.ts

```diff
@@ -0,0 +1,230 @@
+import { ajax, css } from "jquery";
+
+const myClass = new Myclass();
+
+/** Initializes the table head */
+function createTable(): Promise<string[]> {
+	return myClass.fetchColumns()
+		.then(columns => {
+			for (const col of columns) {
+				$(".head").append(`<th>${col}</th>`);
+			}
+			return columns;
+		})
+		.catch(err => {
+			throw 'Error creating table' + err;
+		});
+}
+
+/** calculates the number of rows that can fit the screen */
+const calculatingRows = (): number => {
+	const screenHeight = window.innerHeight;
+	const availableHeight = screenHeight - 105;
+	let rowHeight = 35;
+	if (availableHeight <= 0) {
+		return 0;
+	} else {
+		let maxRows = Math.floor(availableHeight / rowHeight);
+		return maxRows;
+	}
+};
+
+/** calls to re-display records when screen is adjusted */
+function handleResize(recordCount: number) {
+	$(window).on('resize', () => {
+		clearTimeout(myClass.resizeTimeout);
+		myClass.resizeTimeout = setTimeout(() => {
+			displayRecords(recordCount)
+				.then(() => {
+					let inputValue = $('#searchInput').val();
+					if (inputValue !== '') {
+						return searchRecordsAndResize(recordCount);
+					}
+				})
+				.catch(err => {
+					throw new Error('Error occured while resizing' + err);
+				});
+		}, 250);
+	});
+}
+
+/** display records that fit the screen */
+function displayRecords(recordCount: number): Promise<any[]> {
+	$('#loader').show();
+	const inputValue = $('#searchInput').val() as number;
+	const calculatedRows = calculatingRows();
+	const { firstNumber, lastNumber } = calculateFirstAndLastNumbers(calculatedRows, recordCount);
+	updateArrowVisibility(firstNumber, lastNumber, recordCount);
+	return fetchAndDisplayRecords(firstNumber, lastNumber, inputValue)
+		.then(records => {
+			$('#page').empty().append(`Showing record: ${firstNumber} - ${lastNumber}`);
+			$('#loader').hide();
+			return records;
+		}).catch(err => {
+			throw new Error('Error displaying records' + err);
+		});
+}
+
+function calculateFirstAndLastNumbers(calculatedRows: number, recordCount: number) {
+	let firstNumber, lastNumber;
+	if (myClass.firstNumber < 0 || myClass.firstNumber > recordCount) {
+		firstNumber = 0;
+	} else {
+		firstNumber = myClass.firstNumber;
+	}
+	lastNumber = firstNumber + calculatedRows - 1;
+	if (lastNumber >= recordCount) {
+		lastNumber = recordCount;
+		firstNumber = lastNumber - (calculatedRows - 1);
+	}
+	return { firstNumber, lastNumber };
+}
+
+function updateArrowVisibility(firstNumber: number, lastNumber: number, recordCount: number) {
+	if (firstNumber === 0) {
+		$('.arrow-left').hide();
+	} else {
+		$('.arrow-left').show();
+	}
+
+	if (lastNumber >= recordCount) {
+		$('.arrow-right').hide();
+	} else {
+		$('.arrow-right').show();
+	}
+}
+
+function fetchAndDisplayRecords(firstNumber: number, lastNumber: number, inputValue: number) {
+	return myClass.fetchRecords(firstNumber, lastNumber)
+		.then(records => {
+			const tbody = $("tbody");
```

Comment:
I don't think it is a good idea to get every `tbody` on the page, rather change this to use an `id`